sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -325,131 +337,161 @@ static int get_operation_status()
     if ((rc
          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
         != MQTTCLIENT_SUCCESS)
     {
       printf("Failed to receive message, MQTTClient return code %d\n", rc);
       return rc;
     }
-
     printf("Received a message from service.\n");
-    printf("Topic: %s\n", topic);
-
-    if (topic_len == 0)
-    {
-      // The length of the topic if there are one or more NULL characters embedded in topic,
-      // otherwise topic_len is 0.
-      topic_len = (int)strlen(topic);
-    }
-
-    az_span topic_span = az_span_init((uint8_t*)topic, topic_len);
-    az_span message_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
-
-    // Parse the incoming message and payload
-    az_iot_provisioning_client_register_response response;
-    if (az_failed(
-            rc = az_iot_provisioning_client_parse_received_topic_and_payload(
-                &provisioning_client, topic_span, message_span, &response)))
-    {
-      printf("Message from unknown topic, az_result return code %0x4\n", rc);
-      MQTTClient_freeMessage(&message);
-      MQTTClient_free(topic);
-      return rc;
-    }
-
-    printf("Received payload:\n");
-    print_az_span(message_span);
-    printf("Response status is %d.\n", response.status);
 
-    // Parse the operation status from a string to an enum
-    az_iot_provisioning_client_operation_status operation_status;
+    // Parse operation message
     if (az_failed(
-            rc = az_iot_provisioning_client_parse_operation_status(&response, &operation_status)))
+            rc = parse_operation_message(topic, topic_len, message, &response, &operation_status)))
     {
-      printf("Failed to parse operation_status, az_result return code %0x4\n", rc);
-      MQTTClient_freeMessage(&message);
-      MQTTClient_free(topic);
-      return rc;
+      printf("Failed to parse operation message, az_result return code %04x\n", rc);
+      return get_operation_status_free(rc, topic, message);
     }
+    printf("Parsed operation message.\n");
 
-    // Check whether or not the operation is complete
+    // If operation is not complete, send query and loop to receive operation message
     is_operation_complete = az_iot_provisioning_client_operation_complete(operation_status);
     if (!is_operation_complete)
     {
-      // In case the operation is not complete, issue a new query to the service
-      // Get the topic to send the query message
-      if (az_failed(
-              rc = az_iot_provisioning_client_query_status_get_publish_topic(
-                  &provisioning_client, &response, query_topic, sizeof(query_topic), NULL)))
-      {
-        printf("Unable to get query status publish topic, az_result return code %04x\n", rc);
-        MQTTClient_freeMessage(&message);
-        MQTTClient_free(topic);
-        return rc;
-      }
+      get_operation_status_free(0, topic, message);
```

Comment:
Likely a compiler warning (treated as error) as you're ignoring return code.

Reply:
All builds past this one.  good style q tho.